aten/src/ATen/native/cuda/Unique.cu

```diff
@@ -156,20 +156,18 @@ template <typename scalar_t>
 } // namespace
 
 std::tuple<Tensor, Tensor>
-_unique_cuda(const Tensor& self, const bool sorted, const bool return_inverse) {
+_unique_cuda(const Tensor& self, const bool sorted, const bool return_inverse, optional<int64_t> dim) {
+  if (dim) {
+    return AT_DISPATCH_ALL_TYPES(self.type(), "unique", [&] {
+      return _unique_dim_cuda_template<scalar_t>(self, dim.value(), return_inverse);
+    });
+  }
   return AT_DISPATCH_ALL_TYPES(self.type(), "unique", [&] {
     // The current CUDA implementation of unique always sort due to the
     // lack of hashtable implementation in thrust
     return _unique_cuda_template<scalar_t>(self, return_inverse);
   });
 }
 
-std::tuple<Tensor, Tensor>
-_unique_dim_cuda(const Tensor& self, const int64_t dim, const bool sorted, const bool return_inverse) {
```

Comment:
why removing this function? shouldn't cpu and cuda share the same logic?

Reply:
Because `unique_dim` now just calls `unique`, so the logic here is moved to `_unique_cuda`

Reply:
`_unique_dim_cpu` is removed as well